js/add_student.js

```diff
@@ -1,101 +1,226 @@
+        document.getElementById("main").appendChild(document.createElement("br"));
+
+        document.getElementById("main").appendChild(headerDiv);
+
     }
-    return result;
+
+}
 
 
-}//Function to display student information for review
-function dispStudent(){
-    for (let i = 0; i < array[0].length; i++) {
-        document.writeln("<pre>");
-        document.write(sheetHeaders[i] + ": ");
-        document.write(columns[i]);
+function getData() {
+    for(let i = 0; i < uplsHeaders.length; i++) {
+        if(i === 13) {continue;}
+        fields[i] = document.getElementById("field" + i).value;
+
+        console.log(document.getElementById("field"+i).value);
     }
+    objectArray[1] = fields;
+    console.log(fields);
+    var student = confirm("Are you sure you want to add this student?");
+    if(student) {
+        sendData();
+    } else {}
 }
 
-//Function to add a student using API.js functions
-function addStudent(){
-    sa.getTableHeaders("UPLS").then(response => {
-        sheetHeaders = sa.parseTableHeaders(response);
-        sa.insertIntoTableColValues(sheetHeaders, "UPLS", objectArray).then(response => {
-            console.log(sa.parseInsert(response));
-            window.alert("You've added a student");
+function sendData() {
+    objectArray = sa.arrayToObjects(objectArray);
+    sa.insertIntoTableColValues(uplsHeaders, "UPLS", objectArray).then(response => {
+        console.log(sa.parseInsert(response));
+    });
+
+}
+
+function getLibs(){
+    sa.getSheet("LIBRARIANS").then(response => {
+        console.log(response);
+        librarians = sa.parseSheetValues(response);
+        console.log(librarians);
+    });
+}
+
+function getVocab() {
+    sa.getTableHeaders("HEADER_VOCAB_TYPE").then(response => {
+        vocabHeaders = sa.parseTableHeaders(response);
+        sa.getDataType().then(response => {
+            vocabs = sa.parseVocab(response, vocabHeaders);
+            for(let i = 0; i < vocabs.length; i++){
+                vocabsArray[i] = vocabs[i];
+            }
+            console.log(vocabsArray);
+        });
+    });
+}
+function getTypes() {
+    sa.getTableHeaders("HEADER_VOCAB_TYPE").then(response => {
+        vocabHeaders = sa.parseTableHeaders(response);
+        sa.getDataType().then(response => {
+            types = sa.parseDataType(response, vocabHeaders);
+
+            for(let i = 0; i < types.length; i++){
+                typeArray[i] = types[i];
+            }
+            console.log(typeArray);
+            getHeaders();
         });
     });
+}
+
 
+//GET LIST OF PROGRAMS
+function getProgs() {
+    sa.getSheet("PROGRAMS_AND_LIBRARIANS").then(response => {
+        progInfo = sa.parseSheetValues(response);
+        for(let i = 1; i < progInfo.length; i++){
+            if(progInfo[i][1] === lib) {
+                progArray[i] = progInfo[i][0];
+            }
+            if(lib == "upei.personal.librarian@gmail.com"){
```

Comment:
Use the isAdmin() function that Zeru added to SheetsApi.js. We want to avoid hard-coding stuff like this.

Reply:
done